### PR TITLE
BAEL-4436 move email templates to dedicated directory

### PR DESCRIPTION
**Pull request by bcaure**

### CODE REVIEW

File: spring-mvc-basics-2/src/main/java/com/baeldung/spring/configuration/EmailConfiguration.java [hunk 1]
```diff
@@ -2,21 +2,26 @@
...
+import org.thymeleaf.templateresolver.ClassLoaderTemplateResolver;
+import org.thymeleaf.templateresolver.ITemplateResolver;
 
-@Configuration
 @ComponentScan(basePackages = { "com.baeldung.spring.mail" })
+@PropertySource(value={"classpath:application.properties"})
 public class EmailConfiguration {
+	
+	@Value("${mail.templates.path}")
+	String mailTemplatesPath;
```
Comment (ashleyfrieze) [inline spring-mvc-basics-2/src/main/java/com/baeldung/spring/configuration/EmailConfiguration.java:24]: Indentation has gone funny - this should use spaces not tabs

Comment (ashleyfrieze) [inline spring-mvc-basics-2/src/main/java/com/baeldung/spring/configuration/EmailConfiguration.java:24]: This should probably be `private` too

**Review by ashleyfrieze — CHANGES_REQUESTED** [review]

File: spring-mvc-basics-2/src/main/java/com/baeldung/spring/configuration/EmailConfiguration.java (hunk 1 above)
Comment (bcaure) [reply spring-mvc-basics-2/src/main/java/com/baeldung/spring/configuration/EmailConfiguration.java]: indeed, updated in the last commit

Comment (bcaure) [reply spring-mvc-basics-2/src/main/java/com/baeldung/spring/configuration/EmailConfiguration.java]: indeed, updated in the last commit

File: spring-mvc-basics-2/src/main/java/com/baeldung/spring/configuration/EmailConfiguration.java [hunk 2]
```diff
@@ -53,19 +88,28 @@ public SpringTemplateEngine thymeleafTemplateEngine() {
     }
 
     @Bean
-    public SpringResourceTemplateResolver thymeleafTemplateResolver() {
-        SpringResourceTemplateResolver templateResolver = new SpringResourceTemplateResolver();
-        templateResolver.setPrefix("/WEB-INF/views/mail/");
+    public ITemplateResolver thymeleafTemplateResolver() {
+    	AbstractConfigurableTemplateResolver templateResolver;
+    	templateResolver = mailTemplatesExternal ? new FileTemplateResolver() : new ClassLoaderTemplateResolver();
```
Comment (ashleyfrieze) [inline spring-mvc-basics-2/src/main/java/com/baeldung/spring/configuration/EmailConfiguration.java:93]: Declaration of the variable can be on the same line - we don't need a split declaration on L92.

File: spring-mvc-basics-2/src/main/java/com/baeldung/spring/configuration/EmailConfiguration.java [hunk 3]
```diff
@@ -53,19 +88,28 @@ public SpringTemplateEngine thymeleafTemplateEngine() {
...
         return templateResolver;
     }
     
     @Bean 
-    public FreeMarkerConfigurer freemarkerConfig() { 
-        FreeMarkerConfigurer freeMarkerConfigurer = new FreeMarkerConfigurer(); 
-        freeMarkerConfigurer.setTemplateLoaderPath("/WEB-INF/views/mail");
+    public FreeMarkerConfigurer freemarkerConfig() throws IOException {
+        Configuration configuration = new Configuration(Configuration.VERSION_2_3_27);
+        if (mailTemplatesExternal) {
```
Comment (ashleyfrieze) [inline spring-mvc-basics-2/src/main/java/com/baeldung/spring/configuration/EmailConfiguration.java:104]: This seems to go against the discussion where we said we wouldn't make this a configuration option so much as two implementations the reader could choose between.

**Pull request closed without merging**